edb/edgeql/compiler/stmt.py

```diff
@@ -131,6 +132,14 @@ def compile_ForQuery(
             iterator_stmt = setgen.new_set_from_set(
                 iterator_view, preserve_scope_ns=True, ctx=scopectx)
 
+            ptr_target = inference.infer_type(iterator_stmt, ctx.env)
+            anytype = ptr_target.find_any(ctx.env.schema)
+            if anytype is not None:
```

Comment:
Would `ptr_target.is_polymorphic()` work here?

Speaking of `edb/schema/types.py`, `Type.contains_any()` implementation is off w.r.t. its return type.

Reply:
> Would `ptr_target.is_polymorphic()` work here?

It would, but `find_any` gives you the benefit of providing the source context.

> `Type.contains_any()` implementation is off w.r.t. its return type.

How is it wrong? Seems OK to me.

Reply:
> It would, but find_any gives you the benefit of providing the source context.

Got it.

> How is it wrong? Seems OK to me.

My bad, I thought it calls `find_any` instead of `is_any` for some reason.

Reply:
Though we could just use the type itself without any trouble in the cases that seem likely here